cmd/amp/cli/cli_test.go

```diff
@@ -0,0 +1,192 @@
+	}
+	elasticsearchURL = os.Getenv("elasticsearchURL")
+	if elasticsearchURL == "" {
+		elasticsearchURL = elasticsearchDefaultURL
+	}
+	kafkaURL = os.Getenv("kafkaURL")
+	if kafkaURL == "" {
+		kafkaURL = kafkaDefaultURL
+	}
+	influxURL = os.Getenv("influxURL")
+	if influxURL == "" {
+		influxURL = influxDefaultURL
+	}
+	dockerURL = os.Getenv("DOCKER_HOST")
+	if dockerURL == "" {
+		dockerURL = dockerDefaultURL
+	}
+	dockerVersion = os.Getenv("DOCKER_API_VERSION")
+	if dockerVersion == "" {
+		dockerVersion = dockerDefaultVersion
+	}
+	// update config
+	config.Port = port
+	for _, s := range strings.Split(etcdEndpoints, ",") {
+		config.EtcdEndpoints = append(config.EtcdEndpoints, s)
+	}
+	config.ElasticsearchURL = elasticsearchURL
+	config.KafkaURL = kafkaURL
+	config.InfluxURL = influxURL
+	config.DockerURL = dockerURL
+	config.DockerVersion = dockerVersion
+}
+
+// Amplifier
+func TestMain(m *testing.M) {
+	parseEnv()
+	go server.Start(config)
+
+	ctx = context.Background()
+
+	// there is no event when the server starts listening, so we just wait a second
+	time.Sleep(1 * time.Second)
+
+	conn, err := grpc.Dial(serverAddress, grpc.WithInsecure())
+	if err != nil {
+		fmt.Println("connection failure")
+		os.Exit(1)
+	}
+	client = stack.NewStackServiceClient(conn)
+	os.Exit(m.Run())
+}
+
+func TestCmds(t *testing.T) {
+	parseEnv()
+	tests := loadFiles(t)
+	for _, test := range tests {
+		parse(t, test)
+	}
+}
+
+func loadFiles(t *testing.T) []*TestSpec {
+	tests := []*TestSpec{}
+	files, err := ioutil.ReadDir(testDir)
+	if err != nil {
+		t.Error(err)
+		return nil
+	}
+	for _, f := range files {
+		name := f.Name()
+		fmt.Println("Loading file:", name)
+		valid := false
+		if !strings.HasPrefix(name, "invalid-") {
+			valid = true
+		}
+		contents, err := ioutil.ReadFile(path.Join(testDir, name))
+		if err != nil {
+			t.Errorf("unable to load test sample: %s. Error: %v", name, err)
+		}
+		testSpec := &TestSpec{
+			fileName: name,
+			contents: contents,
+			valid:    valid,
+		}
+		tests = append(tests, testSpec)
+	}
+	return tests
+}
+
+func parse(t *testing.T, test *TestSpec) {
+	commandMap, err := parseCommandMap(test.contents)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	for _, spec := range commandMap {
+		cmdSplit := strings.Fields(spec.Cmd)
+		optionsSplit := []string{}
+		for _, val := range spec.Options {
+			optionsSplit = append(optionsSplit, strings.Fields(val)...)
+		}
+		commandFinal := append(cmdSplit, spec.Args...)
+		commandFinal = append(commandFinal, optionsSplit...)
+		fmt.Println(commandFinal, "Command passed")
+		RunCmd(t, commandFinal)
+	}
+}
+
+func RunCmd(t *testing.T, cmdString []string) {
+	fmt.Println(cmdString, "Running...")
+	cmd := exec.Command(cmdString[0], cmdString[1:]...)
+	result, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Log(cmdString)
+		t.Error(err)
+		t.Fail()
+	} else {
+		fmt.Print(cmdString, " Command result:\n", string(result))
+	}
+	time.Sleep(2 * time.Second)
```

Comment:
Sleep call removed. Replaced with a for loop for recursing a command that fails. Primarly for the purpose of ensure that the pinger isn't curled before pinger has appeared in the swarm.